batchlib/analysis/cell_level_analysis.py

```diff
@@ -767,11 +802,25 @@ def write_well_table(self, input_files):
                                                                              image_names_for_current_well)
                                          if not image_outlier_dict.get(im_name, (-1, ''))[0] == 1]
             n_outlier_images = n_total - len(in_files_for_current_well)
-            if len(in_files_for_current_well) == 0:
-                # TODO: add row full of np.nan for wells of outliers
-                logger.info(f'Skipping well {well_name} as it consists entirely of outliers. '
-                            f'It will not be included in the per-well table.')
+
+            well_is_outlier, outlier_type = well_outlier_dict.get(well_name, (-1, 'not checked'))
+            if well_is_outlier:
+                logger.info(f'Well {well_name} was flagged as outlier.')
+
+            if (well_is_outlier != 1) and (len(in_files_for_current_well) == 0):
+                logger.info(f'Well {well_name} consists entirely of outlier images and will be marked as outlier well.')
+                well_is_outlier = 1
+                outlier_type = 'all images are outliers'
+
+            # add row of nan's if this well is an outlier
+            if well_is_outlier == 1:
+                n_stats = len(column_names) - 5
+                # FIXME in the corner case where the first well we encounter is an outlier, this will fail!
+                # @Roman, what's the best way to fix this?
+                assert n_stats > 0, "Firs well was outlier"
+                table.append([well_name, n_outlier_images, np.nan, well_is_outlier, outlier_type] + [np.nan] * n_stats)
                 continue
```

Comment:
I would just append `None` to the table here. Then get `n_stats` after the for loop and replace all the `None`s in the table with rows full of `np.nan`.

Reply:
ok, that makes a lot of sense, I will implement it.

Reply:
Ok, I simplified it a bit more.
I am not sure if this catches all corner cases (in the case where all input images are outliers),
but I will just write a test for this and we can fix it if it's an issue.